neo/Network/RPC/TransactionManager.cs

```diff
@@ -0,0 +1,220 @@
+        /// <param name="sender">the account script hash of sender</param>
+        public TransactionManager(RpcClient rpc, UInt160 sender)
+        {
+            rpcClient = rpc;
+            this.sender = sender;
+        }
+
+        /// <summary>
+        /// Create an unsigned Transaction object with given parameters.
+        /// </summary>
+        /// <param name="script">Transaction Script</param>
+        /// <param name="attributes">Transaction Attributes</param>
+        /// <param name="cosigners">Transaction Cosigners</param>
+        /// <param name="networkFee">Transaction NetworkFee, will set to estimate value(with only basic signatures) when networkFee is 0</param>
+        /// <returns></returns>
+        public TransactionManager MakeTransaction(byte[] script, TransactionAttribute[] attributes = null, Cosigner[] cosigners = null, long networkFee = 0)
+        {
+            uint height = rpcClient.GetBlockCount() - 1;
+            Tx = new Transaction
+            {
+                Version = 0,
+                Nonce = (uint)rand.Next(),
+                Script = script,
+                Sender = sender,
+                ValidUntilBlock = height + Transaction.MaxValidUntilBlockIncrement,
+                Attributes = attributes ?? new TransactionAttribute[0],
+                Cosigners = cosigners ?? new Cosigner[0],
+                Witnesses = new Witness[0]
+            };
+
+            RpcInvokeResult result = rpcClient.InvokeScript(script);
+            Tx.SystemFee = Math.Max(long.Parse(result.GasConsumed) - ApplicationEngine.GasFree, 0);
+            if (Tx.SystemFee > 0)
+            {
+                long d = (long)NativeContract.GAS.Factor;
+                long remainder = Tx.SystemFee % d;
+                if (remainder > 0)
+                    Tx.SystemFee += d - remainder;
+                else if (remainder < 0)
+                    Tx.SystemFee -= remainder;
+            }
+
+            context = new ContractParametersContext(Tx);
+
+            // set networkfee to estimate value when networkFee is 0
+            Tx.NetworkFee = networkFee == 0 ? EstimateNetworkFee() : networkFee;
+
+            var gasBalance = new Nep5API(rpcClient).BalanceOf(NativeContract.GAS.Hash, sender);
+            if (gasBalance >= Tx.SystemFee + Tx.NetworkFee) return this;
+            throw new InvalidOperationException($"Insufficient GAS in address: {sender.ToAddress()}");
+        }
+
+        /// <summary>
+        /// Estimate NetworkFee, assuming the witnesses are basic Signature Contract
+        /// </summary>
+        private long EstimateNetworkFee()
+        {
+            long networkFee = 0;
+            UInt160[] hashes = Tx.GetScriptHashesForVerifying(null);
+            int size = Transaction.HeaderSize + Tx.Attributes.GetVarSize() + Tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
+
+            // assume the hashes are single Signature
```

Comment:
If is multi signature? because is possible to add multi signature, but the `EstimateNetworkFee` seems not compatible

Reply:
This method is used before we sign the transaction and only works when the input network fee is Zero, so we just consider the single signature in this case.